blobstore/blobstore.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+class BlobStore(object):
+    """Abstract base class for all blob stores."""
+    def __init__(self):
+        pass
+
+    def list(self, prefix=None):
+        """Returns a list of all blob entries in a container that match a given prefix."""
+        raise NotImplementedError()
+
+    def get_presigned_url(self, objname):
+        """Retrieves a presigned url for blob objname.  Raises BlobNotFoundError if the blob is not present."""
+        raise NotImplementedError()
+
+    def set(self, objname, src_file_handle):
```

Comment:
I didn't realize there was a presigned upload endpoint. Definitely will be tweaking this as I gain better understanding of what's available on S3...